packages/core/build/tokens.ts

```diff
@@ -6,43 +6,62 @@
 
 import { token, CdsTheme } from './token-utils';
 
+// internal optimization so tokens can refer to pre-calculated values rather than generating/duplicating the same calc over many tokens
+const internal = {
+  scale1: token('calc((1rem / var(--cds-global-base)) * var(--cds-global-scale-layout-space))', { raw: true }),
+  scale2: token('calc((1rem / var(--cds-global-base)) * var(--cds-global-scale-space))', { raw: true }),
+  scale3: token('calc((1rem / var(--cds-global-base)) * var(--cds-global-scale-typography))', { raw: true }),
+};
+
+const base = token(20, { raw: true });
+
+const scale = {
+  space: token(1, { raw: true }),
+  layoutSpace: token(1, { raw: true }),
+  typography: token(1, { raw: true }),
+};
+
 const layout = {
-  space: {
-    xxs: token(2),
-    xs: token(4),
-    sm: token(6),
-    md: token(12),
-    lg: token(24),
-    xl: token(48),
-    xxl: token(96),
-  },
   grid: {
-    cols: token(12, { static: true }),
+    cols: token(12, { raw: true }),
   },
   width: {
-    xs: token('576px', { static: true }),
-    sm: token('768px', { static: true }),
-    md: token('992px', { static: true }),
-    lg: token('1200px', { static: true }),
-    xl: token('1440px', { static: true }),
+    xs: token('576px', { raw: true, static: true }),
+    sm: token('768px', { raw: true, static: true }),
+    md: token('992px', { raw: true, static: true }),
+    lg: token('1200px', { raw: true, static: true }),
+    xl: token('1440px', { raw: true, static: true }),
+  },
+  space: {
+    xxxs: token(2, { scale: internal.scale1 }),
+    xxs: token(4, { scale: internal.scale1 }),
+    xs: token(8, { scale: internal.scale1 }),
+    sm: token(12, { scale: internal.scale1 }),
+    md: token(16, { scale: internal.scale1 }),
+    lg: token(24, { scale: internal.scale1 }),
+    xl: token(32, { scale: internal.scale1 }),
+    xxl: token(48, { scale: internal.scale1 }),
+    xxxl: token(64, { scale: internal.scale1 }),
   },
 };
 
 const space = {
-  0: token(0),
-  1: token(1),
-  2: token(2),
-  3: token(4),
-  4: token(6),
-  5: token(8),
-  6: token(12),
-  7: token(16),
-  8: token(18),
-  9: token(24),
-  10: token(32),
-  11: token(36),
-  12: token(48),
-  13: token(72),
+  0: token(0, { scale: internal.scale2 }),
+  1: token(1, { scale: internal.scale2 }),
+  2: token(2, { scale: internal.scale2 }),
+  3: token(4, { scale: internal.scale2 }),
+  4: token(6, { scale: internal.scale2 }),
+  5: token(8, { scale: internal.scale2 }),
+  6: token(12, { scale: internal.scale2 }),
+  7: token(16, { scale: internal.scale2 }),
+  8: token(18, { scale: internal.scale2 }),
+  9: token(24, { scale: internal.scale2 }),
+  10: token(32, { scale: internal.scale2 }),
+  11: token(36, { scale: internal.scale2 }),
+  12: token(48, { scale: internal.scale2 }),
+  13: token(64, { scale: internal.scale2 }),
+  14: token(72, { scale: internal.scale2 }),
+  15: token(96, { scale: internal.scale2 }),
```

Comment:
The internal scale alias names are kept short so we don't generate excessively large css vars for non public facing APIs.